drivers/adxl372/adxl372.c

```diff
@@ -772,12 +808,14 @@ int32_t adxl372_init(adxl372_dev **device,
 
 	*device = dev;
```

Comment:
maybe move this under `if (!ret)`  ; otherwise you coudl return with a pointer to something that was free'd